Towards new release 0.22.0

Time flies and we would like to release more often than in the past so lets consider this a draft of release notes for the next release.

Please, let me know if I missed something or if there is something what should go in before release.
@OpenSC/core, @OpenSC/maintainers.

---

### General improvements
 * Use standard paths for file cache on Linux (#2148) and OSX (#2214)
 * Various issues of memory/buffer handling in legacy drivers mostly reported by oss-fuzz and coverity (tcos, oberthur, isoapplet, iasecc, westcos, gpk, flex, dnie, mcrd, authentic, belpic)
 * Add threading test to `pkcs11-tool` (#2067)
 * Add support to generate generic secret keys (#2140)
 * `opensc-explorer`: Print information about LCS (Life cycle status byte) (#2195)
 * Add support for Apple's arm64 (M1) binaries, removed TokenD. A seperate installer with TokenD (and without arm64 binaries) will be available (#2179).
 * Support for gcc11 and its new strict aliasing rules (#2241, #2260)
 * Initial support for building with OpenSSL 3.0 (#2343)
 * pkcs15-tool: Write data objects in binary mode (#2324)

### PKCS#11
 * Support for ECDSA verification (#2211)
 * Support for ECDSA with different SHA hashes (#2190)
 * Prevent issues in p11-kit by not returning unexpected return codes (#2207)
 * Add support for PKCS#11 3.0: The new interfaces, profile objects and functions (#2096, #2293)
 * Standardize the version 2 on 2.20 in the code (#2096)
 * Fix CKA_MODIFIABLE and CKA_EXTRACTABLE  (#2176)
 * Copy arguments of C_Initialize (#2350)

### Minidriver
 * Fix RSA-PSS signing (#2234)

### OpenPGP
 * Fix DO deletion (#2215)
 * Add support for (X)EdDSA keys (#1960)

### IDPrime
 * Add support for applet version 3 and fix RSA-PSS mechanisms (#2205)
 * Add support for applet version 4 (#2332)

### MyEID
 * New configuration option for opensc.conf to disable pkcs1_padding (#2193)
 * Add support for ECDSA with different hashes (#2190)
 * Enable more mechanisms (#2178)
 * Fixed asking for a user pin when formatting a card (#1737)

### IAS/ECC
 * Added support for French CPx Healthcare cards (#2217)

### CardOS
* Added ATR for new CardOS 5.4 version (#2296)
---

outcome on Ubuntu 20.04:
  - https://github.com/OpenSC/OpenSC/archive/refs/tags/0.22.0-rc1.tar.gz : OK, it compiles
  - https://github.com/OpenSC/OpenSC/releases/download/0.22.0-rc1/opensc-0.22.0.tar.gz does not compile, because of:
```
pkcs15-iasecc.c:37:10: fatal error: ../pkcs15init/pkcs15-iasecc.h: No such file or directory
   37 | #include "../pkcs15init/pkcs15-iasecc.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

It seems it was missing from the packaging.

--- a/NEWS
+++ b/NEWS
@@ -1,6 +1,6 @@
 NEWS for OpenSC -- History of user visible changes
 
-# New in 0.22.0; 2021-04-XX
+# New in 0.22.0; 2021-06-XX
 ## General improvements
  * Use standard paths for file cache on Linux (#2148) and OSX (#2214)
  * Various issues of memory/buffer handling in legacy drivers mostly reported by oss-fuzz and coverity (tcos, oberthur, isoapplet, iasecc, westcos, gpk, flex, dnie, mcrd, authentic, belpic)
@@ -9,13 +9,16 @@ NEWS for OpenSC -- History of user visible changes
  * `opensc-explorer`: Print information about LCS (Life cycle status byte) (#2195)
  * Add support for Apple's arm64 (M1) binaries, removed TokenD. A seperate installer with TokenD (and without arm64 binaries) will be available (#2179).
  * Support for gcc11 and its new strict aliasing rules (#2241, #2260)
+ * Initial support for building with OpenSSL 3.0 (#2343)
+ * pkcs15-tool: Write data objects in binary mode (#2324)
 ## PKCS#11
  * Support for ECDSA verification (#2211)
  * Support for ECDSA with different SHA hashes (#2190)
  * Prevent issues in p11-kit by not returning unexpected return codes (#2207)
- * Add support for PKCS#11 3.0: The new interfaces, profile objects and functions (#2096)
+ * Add support for PKCS#11 3.0: The new interfaces, profile objects and functions (#2096, #2293)
  * Standardize the version 2 on 2.20 in the code (#2096)
  * Fix CKA_MODIFIABLE and CKA_EXTRACTABLE  (#2176)
+ * Copy arguments of C_Initialize (#2350)
 ## Minidriver
  * Fix RSA-PSS signing (#2234)
 ## OpenPGP
@@ -23,6 +26,7 @@ NEWS for OpenSC -- History of user visible changes
  * Add support for (X)EdDSA keys (#1960)
 ## IDPrime
  * Add support for applet version 3 and fix RSA-PSS mechanisms (#2205)
+ * Add support for applet version 4 (#2332)
 ## MyEID
  * New configuration option for opensc.conf to disable pkcs1_padding (#2193)
  * Add support for ECDSA with different hashes (#2190)

